interfaces/matlab_experimental/1D/Domain1D.m

```diff
@@ -0,0 +1,696 @@
+        %     True if the domain is a surface, and false otherwise.
+        isSurface
+
+        % Mass flux. ::
+        %
+        %     >> mdot = d.massFlux
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     The mass flux in the domain.
+        massFlux
+
+        % Number of components. ::
+        %
+        %     >> n = d.nComponents
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Number of variables at each grid point.
+        nComponents
+
+        % Get the number of grid points. ::
+        %
+        %     >> n = d.nPoints
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Integer number of grid points.
+        nPoints
+
+    end
+
+    methods
+        %% Domain1D Class Constructor.
+
+        function d = Domain1D(varargin)
+            % Create a :mat:class:`Domain1D` object. 
+
+            ctIsLoaded;
+
+            if nargin == 1
+                a = varargin{1};
+
+                if strcmp(a, 'Inlet1D')
+                    d.domainID = ctFunc('inlet_new');
+                elseif strcmp(a, 'Surf1D')
+                    d.domainID = ctFunc('surf_new');
+                elseif strcmp(a, 'Symm1D')
+                    d.domainID = ctFunc('symm_new');
+                elseif strcmp(a, 'Outlet1D')
+                    d.domainID = ctFunc('outlet_new');
+                elseif strcmp(a, 'OutletRes')
+                    d.domainID = ctFunc('outletres_new');
+                else
+                    error('Not enough arguments for that job number');
+                end
+
+            elseif nargin == 2
+                a = varargin{1};
+                b = varargin{2};
+
+                if strcmp(a, 'StagnationFlow')
+
+                    if isa(b, 'Solution')
+                        d.domainID = ctFunc('stflow_new', b.tpID, b.kinID, b.trID, 1);
+                    else
+                        error('Wrong argument type. Expecting instance of class Solution.');
+                    end
+
+                elseif strcmp(a, 'AxisymmetricFlow')
+
+                    if isa(b, 'Solution')
+                        d.domainID = ctFunc('stflow_new', b.tpID, b.kinID, b.trID, 2);
+                    else
+                        error('Wrong argument type. Expecting instance of class Solution.');
+                    end
+
+                elseif strcmp(a, 'ReactingSurface')
+
+                    if isa(b, 'Interface')
+                        d.domainID = ctFunc('reactingsurf_new');
+                        ctFunc('reactingsurf_setkineticsmgr', d.domainID, b.kinID);
+                    else
+                        error('Wrong argument type. Expecting instance of class Interface.');
+                    end
+
+                else
+                    error('Wrong object type.');
+                end
```

Comment:
same as above. All exception handling is already being handled by the C++ layer.